halo2-ecc/src/ecc/mod.rs

```diff
@@ -1014,6 +1036,36 @@ impl<'chip, F: BigPrimeField, FC: FieldChip<F>> EccChip<'chip, F, FC>
 where
     FC: Selectable<F, FC::FieldPoint>,
 {
+    /// Expensive version of `sum_unsafe`, but works generally
+    pub fn sum<C>(
+        &self,
+        ctx: &mut Context<F>,
+        points: impl IntoIterator<Item = EcPoint<F, FC::FieldPoint>>,
+    ) -> EcPoint<F, FC::FieldPoint>
+    where
+        C: CurveAffineExt<Base = FC::FieldType>,
+    {
+        let rand_point = self.load_random_point::<C>(ctx);
+        let rand_point2 = self.load_random_point::<C>(ctx);
+        let zero = ctx.load_constant(F::ZERO);
+        let neg_rand_point = self.negate(ctx, rand_point.clone());
+        let mut acc = ComparableEcPoint::from(neg_rand_point.clone());
+        for point in points {
+            let point_is_inf = self.is_infinity(ctx, point.clone());
+            let addend = self.select(ctx, rand_point2.clone(), point.clone(), point_is_inf);
```

Comment:
there is perhaps a small optimization here that your `is_infinity` check already makes `point` a `StrictEcPoint`, so then `addend` can already be assumed `Strict`. OK to not do this optimization.

separately `is_infinity` could be optimized if the input assumes `point` is zero only if `point.x = 0` and `point.y = 0` as bigint, not just as field elements. Again, OK to leave as is.

Reply:
Confused about the StrictEcPoint stuff - where does this occur?